R/add_subprotocol.R

```diff
@@ -93,22 +82,26 @@ add_subprotocol <-
       file.path(file_name) %>%
       path_rel(start = find_root(is_git_root))
 
+    firstremote <- execshell("git remote", intern = TRUE)[1]
+    execshell(paste0("git fetch ", firstremote),
+              ignore.stdout = TRUE,
+              ignore.stderr = TRUE)
```

Comment:
This will break the tests in `test-add-subprotocol.R`. For testing, it is easier if we can just pretend that everything is local (no remote). I agree that in real use cases fetching the remote is a useful extra step (in case the user of the function forgot to that in RStudio). Maybe we can add an argument `fetch_remote = TRUE` and make these lines conditional? That would make it easy for me to adapt `test-add-subprotocol.R` so that everything works.

Reply:
> Maybe we can add an argument `fetch_remote = TRUE` and make these lines conditional

Good suggestion. Limitations in testing should not be a reason to decrease functionality. So having `fetch_remote = TRUE` as default but `FALSE` for the tests would be OK.